bin/lisky.js

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env node
+/*
+ * LiskHQ/lisky
+ * Copyright © 2017 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+import os from 'os';
+import lockfile from 'lockfile';
+import semver from 'semver';
+// eslint-disable-next-line import/order
+import packageJSON from '../package.json';
+
+const nonInteractiveLiskyArg = process.argv[1];
+const nonInteractiveCommandArg = process.argv[2];
+
+process.env.LISKY_CONFIG_DIR =
+	process.env.LISKY_CONFIG_DIR || `${os.homedir()}/.lisky`;
+const configLockfilePath = `${process.env.LISKY_CONFIG_DIR}/config.lock`;
+
+process.env.NON_INTERACTIVE_MODE = !(
+	nonInteractiveLiskyArg.endsWith('lisky') && process.argv.length === 2
+);
+
+function exit(code) {
+	process.exit(code || 0);
+}
+
+if (!semver.satisfies(process.version, packageJSON.engines.node)) {
+	console.error(
+		'\x1b[31m',
+		`ERROR: Requires Node.js version ${semver.clean(
+			packageJSON.engines.node,
+		)}, but was started with version ${semver.clean(process.version)}.`,
+		'\x1b[0m',
+	);
+	exit();
+}
+
+switch (process.argv[2]) {
+	case 'clean':
+		console.warn(
+			'\x1b[33m',
+			'WARNING: Attempting to remove configuration lockfile. I hope you know what you’re doing.',
+			'\x1b[0m',
+		);
+		lockfile.unlockSync(configLockfilePath);
+		exit();
+		break;
+	case '--version':
+	case '-v':
+		console.info(packageJSON.version);
+		exit();
+		break;
+	default:
+	// continue...
+}
+
+// eslint-disable-next-line import/extensions,import/no-unresolved
+const lisky = require('../dist').default;
+// eslint-disable-next-line import/extensions,import/no-unresolved
```

Comment:
Do we want these? Prefer a space after the comma if we do.

Reply:
We need this, since before the build it will complain.
Also, space after is not permitted by other `import/order` rule

Reply:
The other rule here is `import/extensions`, which seems irrelevant. If we get rid of `import/no-unresolved` we'll miss issues if e.g. the filename for `dist/exec_file.js` changes.

Reply:
Yes `import/extensions` is unnecessary.
> If we get rid of import/no-unresolved we'll miss issues if e.g. the filename for dist/exec_file.js changes.
This is true, and it's already happened. However, considering CI is always fresh build, it will not pass...

Reply:
We could just build before linting?

Reply:
I think usually lint should come first since it's more basic check than build, but in this specific case, i think it's okay if there are no alternative.

Reply:
Alternatively we could have two different lint scripts, one for `src` and one for `bin`, and only run the `bin` one after building?